src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,106 +69,233 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
+        selectedItem.current.scrollIntoView();
+      }
+      previousSelectedPath.current = selectedPath;
     }
-  });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [openKeys]);
+
+  /**
+   * Ensures the parent of the currently selected item is expanded.
+   */
+  useEffect(() => {
+    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPath]);
 
   const handleOnClick = (event, item) => {
+    // Don't update state if function is called through handleKeyDown
+    if (event.type === 'click') {
+      const eventTargetId = event.target.getAttribute('id');
+      currentNodeId.current = eventTargetId;
+      cursor.current = visibleNodes.findIndex((el) => el.id === eventTargetId);
+      focusCurrentNode();
+    }
     if (!item.childItems) {
       onSelect(item.path);
       return;
     }
     setOpenKeys({ ...openKeys, [item.path]: !openKeys[item.path] });
   };
 
+  const handleDownArrow = () => {
+    if (cursor.current + 1 < visibleNodes.length) {
+      cursor.current += 1;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  const handleUpArrow = () => {
+    if (cursor.current >= 1) {
+      cursor.current -= 1;
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  /**
+   * Finds parent of the current node, used for left arrow functionality
+   */
+  const findParentNode = () => {
+    if (!currentNodeId.current) {
+      return;
+    }
+    const parentId = visibleNodes.find((el) => el.id === currentNodeId.current).parent;
+    if (parentId !== '') {
+      cursor.current = visibleNodes.findIndex((el) => el.id === parentId);
+      currentNodeId.current = visibleNodes[cursor.current].id;
+      focusCurrentNode();
+    }
+  };
+
+  /**
+   * Finds the first node starting with the given character.
+   * Starts at the cursors current position, and wraps around to the beginning of the menu if a match isn't found.
+   * @param {*} char The character to search by
+   */
+  const findNodeMatching = (char) => {
+    let sortedNodes = visibleNodes.slice(cursor.current + 1, visibleNodes.length);
+    sortedNodes = sortedNodes.concat(visibleNodes.slice(0, cursor.current));
+    const match = sortedNodes.find((el) => el.id[0].toUpperCase() === char);
+    if (match) {
+      cursor.current = visibleNodes.findIndex((el) => el.id === match.id);
+      currentNodeId.current = match.id;
+      focusCurrentNode();
+    }
+  };
+
   const handleKeyDown = (event, item) => {
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      event.preventDefault();
-      handleOnClick(event, item);
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_SPACE:
+      case KeyCode.KEY_RETURN:
+        event.preventDefault();
+        handleOnClick(event, item);
```

Comment:
Holding down space or return while on a node with sub items will repeatedly open and close the nav tree. I think this is expected behavior and a strange edge case. This is something that happens often with buttons too. Unless others disagree I wouldn't focus to much on trying to prevent this from happening, I just thought it was interesting.